blocks/sign-up/sign-up.js

```diff
@@ -0,0 +1,130 @@
+    // TBD : need ajax call for otp generation etc
+  }
+};
+
+/**
+ * Removes non-numeric characters from the input value.
+ * @param {Event} event - The event object triggered by the input element.
+ */
+function blockNonNumbers(event) {
+  const inputElement = event.target;
+  inputElement.value = inputElement.value.replace(/[^0-9]/g, '');
+}
+
+function createMobileNumberInput(placeholderText) {
+  const inputElement = createElement('input', 'phonenumber-textbox');
+  inputElement.type = 'text';
+  inputElement.placeholder = placeholderText;
+  inputElement.maxLength = '10';
+  inputElement.pattern = '[0-9]*';
+  inputElement.autocomplete = 'off';
+  inputElement.addEventListener('input', blockNonNumbers);
+  return inputElement;
+}
+
+function createSubmitButton(buttontitle) {
+  const submitButton = createElement('button', 'signupbtn');
+  submitButton.textContent = buttontitle;
+  submitButton.addEventListener('click', handleOpenAccountSubmit);
+  return submitButton;
+}
+
+function createErrorSpan(errormessage) {
+  const spanElement = createElement('span', 'error-message');
+  spanElement.textContent = errormessage;
+  return spanElement;
+}
+
+function createTitle(titleHTML) {
+  const col1Div = createElement('div', 'signupsections');
+  const titleWrapDiv = createElement('div', 'title-wrap');
+  const h2Element = createElement('h2', '');
+  h2Element.innerHTML = titleHTML.innerHTML;
+  h2Element.classList.add('text-left');
+  titleWrapDiv.appendChild(h2Element);
+  col1Div.appendChild(titleWrapDiv);
+  return col1Div;
+}
+
+function createSignUpElement(
+  signupString,
+  promotionaltext,
+  placeholderText,
+  buttontitle,
+  errormessage,
+) {
+  const signupFormDiv = createElement('div', 'signupsections');
+
+  const formGroupDiv = createElement('div', 'signup-form-group');
+  formGroupDiv.classList.add('text-center');
+
+  const signupStringElement = createElement('label', '');
+  signupStringElement.innerHTML = signupString.innerHTML;
+
+  const promotionalSpan = createElement('span', 'promotional-text');
+  promotionalSpan.textContent = promotionaltext;
+
+  const formFieldsDiv = createElement('div', '');
+  const mobileInput = createMobileNumberInput(placeholderText);
+  const submitButton = createSubmitButton(buttontitle, errormessage);
+  const errorSpan = createErrorSpan(errormessage);
+
+  formFieldsDiv.appendChild(mobileInput);
+  formFieldsDiv.appendChild(submitButton);
+
+  formGroupDiv.appendChild(signupStringElement);
+  formGroupDiv.appendChild(promotionalSpan);
+  formGroupDiv.appendChild(formFieldsDiv);
+  formGroupDiv.appendChild(errorSpan);
+
+  signupFormDiv.appendChild(formGroupDiv);
+  return signupFormDiv;
+}
+
+export default async function decorate(block) {
+  const blockConfig = readBlockConfig(block);
+  const {
+    promotionaltext, placeholdertext, buttontitle, errormessage,
+  } = blockConfig;
+  const titleHTML = block.querySelectorAll(':scope > div')[0].children[1];
```

Comment:
Would it make sense to make it a little resilient by iterating over all children and looking for children[0] to match `title` as value? We can do it here itself in this block code.

Reply:
@vivesing you are right , but we decided to update readblock config to have a default class based on key to div.
that has been tracked https://github.com/aemsites/icicidirect/issues/119. 
I will create a PR for the same. Till then we can keep this implementation as done in another blocks as well.